File: src/NubipApp.Models/Domain/Auditorium.cs
using System.Collections.Generic;

namespace NubipApp.Models.Domain
{
    public class Auditorium
    {
        public int AuditoriumId { get; set; }
        public int Number { get; set; }
        public int BuildingId { get; set; }
        public int SeatseCount { get; set; }
        public int AuditoriumTypeId { get; set; }

        public AuditoriumType AuditoriumType { get; set; }
        public Building Building { get; set; }
        
        public ICollection<Lesson> Lessons { get; set; }
    }
}

File: src/models/auditorium.ts
import mongoose from "mongoose";

let Schema = mongoose.Schema;

let AuditoriumSchema = new Schema({
	nameId: Number,
	number: Number,
	buildingId: [{
		type: Schema.Types.ObjectId, ref: 'Building'
	}],
	searseCount: Number,
	auditoriumTypeId: [{
		type: Schema.Types.ObjectId, ref: 'AuditoriumType'
	}]
});

let Auditorium = mongoose.model('Auditorium', AuditoriumSchema);

export default Auditorium;